### PR TITLE
recommended endpoint

### PR DESCRIPTION
**Pull request by ihorhladkov**

### CODE REVIEW

File: src/controllers/product.controller.ts [hunk 1]
```diff
@@ -62,4 +63,31 @@ const getFiltered = async (
...
+    const foundProduct = await productsService.getOne(productId);
+
+    if (foundProduct) {
+      const allProducts = await Product.find()
+        .populate('description')
+        .populate('category');
+
+      const randomProducts = productsService.getRandom(allProducts, 10);
+
+      res.status(200).json(randomProducts);
```
Comment (olena-ole) [inline src/controllers/product.controller.ts]:
```suggestion
      res.status(200).send(randomProducts);
```

File: src/controllers/product.controller.ts [hunk 2]
```diff
@@ -62,4 +63,31 @@ const getFiltered = async (
...
+
+  try {
+    const foundProduct = await productsService.getOne(productId);
+
+    if (foundProduct) {
+      const allProducts = await Product.find()
+        .populate('description')
+        .populate('category');
+
+      const randomProducts = productsService.getRandom(allProducts, 10);
```
Comment (olena-ole) [inline src/controllers/product.controller.ts]: This logic must be in service, Max what do you think? I am not 100% sure

**Review by olena-ole — CHANGES_REQUESTED** [review]

File: src/controllers/product.controller.ts (hunk 1 above)
Comment (MaxSchmide) [reply src/controllers/product.controller.ts]: agree

File: src/controllers/product.controller.ts (hunk 2 above)
Comment (MaxSchmide) [reply src/controllers/product.controller.ts]: yes but then we need new service for all products without any queries or params, so we can reuse it

File: src/controllers/product.controller.ts [hunk 3]
```diff
@@ -62,4 +62,29 @@ const getFiltered = async (
...
+const getRecommended = async (req: Request, res: Response) => {
+  const productId = req.params.id;
+
+  try {
+    const foundProduct = await productsService.getOne(productId);
+
+    if (foundProduct) {
+      const allProducts = await productsService.allProducts();
+
+      const randomProducts = productsService.getRandom(allProducts, 10);
```
Comment (olena-ole) [inline src/controllers/product.controller.ts]: I would move getting all products to service, in controller you only receive the random products
```suggestion
      const randomProducts = await productsService.getRandom(10);
```

File: src/services/product.service.ts [hunk 4]
```diff
@@ -46,4 +54,30 @@ const getFiltered = async (query: string) => {
   return products;
 };
 
-export default { getAll, getOne, getFiltered };
+function getRandom(products: IProduct[], limit: number) {
+  const randomProducts = [];
```
Comment (olena-ole) [inline src/services/product.service.ts]:
```suggestion
function async getRandom(limit: number) {
  const products = await allProducts();
  const randomProducts = [];
```

File: src/services/product.service.ts [hunk 5]
```diff
@@ -46,4 +54,30 @@ const getFiltered = async (query: string) => {
...
+      i++;
+      randomProducts.push(products[randomIndex]);
+    }
+  }
+
+  return randomProducts;
+}
+
+export default {
+  allProducts,
```
Comment (olena-ole) [inline src/services/product.service.ts:77]: You won't need to export allProducts, it will be used only inside service

**Review by olena-ole — CHANGES_REQUESTED** [review]

**Review by olena-ole — APPROVED** [review]

File: src/services/product.service.ts (hunk 5 above)
Comment (MaxSchmide) [reply src/services/product.service.ts:77]: let it be for now

**Review by MaxSchmide — APPROVED** [review]

**Pull request merged**